### ISSUE TITLE
[GSoC] Avoid registering tests datasets

### ISSUE THREAD

**Issue by Conchylicultor** [tensorflow/datasets#1830]
Currently `tfds.list_builder` list [`dummy_mnist`](https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/testing/test_utils.py#L393) and other test-only datasets. We should make sure to avoid registering those datasets.

* Add `with tfds.core.registered.skip_registeration():` context manager which when applied skip dataset registration. Changes should go in `registered.py`. Add tests in `registered_test.py`.
* Apply this to the public API: https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/public_api.py#L24

```
with core.registered.skip_registeration():
  from tensorflow_datasets import testing
```

* Add a `test_dataset_registered.py` which check that `tfds.list_builder` do not contains `dummy_mnist` or other test datasets.

**Comment by Eshan-Agarwal** [comment]
Is test only include dataset with prefix `dummy `is avoid?

**Comment by Conchylicultor** [comment]
@Eshan-Agarwal No, this should works for any datasets, no matter which name it is.

Registration is done in https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/core/registered.py#L148

**Comment by vijayphoenix** [comment]
@Conchylicultor Please review https://github.com/tensorflow/datasets/pull/1832
Thank you

**Comment by Conchylicultor** [comment]
This should be fixed thanks to #1833
